packages/block-editor/src/hooks/gap.js

```diff
@@ -109,6 +113,19 @@ export function GapEdit( props ) {
 		setAttributes( {
 			style: cleanEmptyObject( newStyle ),
 		} );
+
+		// In Safari, changing the `gap` CSS value on its own will not trigger the layout
+		// to be recalculated / re-rendered. To force the updated gap to re-render, here
+		// we replace the block's node with itself.
+		const isSafari =
+			window?.navigator.userAgent &&
+			window.navigator.userAgent.includes( 'Safari' ) &&
+			! window.navigator.userAgent.includes( 'Chrome ' ) &&
+			! window.navigator.userAgent.includes( 'Chromium ' );
+
+		if ( ref.current && isSafari ) {
```

Comment:
I'm debugging another issue but noticed here that with my userAgent:
`"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36"` this returns true.

I'm using Chrome `Version 96.0.4664.110 (Official Build) (x86_64)`

Just mentioning this if it's something we should look into..

Reply:
Thanks @ntsekouras! Unless it's causing other issues, it shouldn't matter too much if the check is greedier than we'd expected — the trade-off is that when this check passes, then the browser is doing more DOM manipulation than might be required for the task at hand, which fixed an issue in Safari, but wasn't generally required in Chrome, Firefox, etc.

If you notice that performance of sliding the block gap control in Chrome is sluggish, however, (or the control is generally buggy) we might want to pursue an optimisation, but personally I'm not too concerned if `replaceChild` is happening in browsers other than Safari.

Reply:
Sure thing! It wasn't anything noticeable at all. I just noticed that during debugging something else 😄